elastic/kibana
# [Remote Clusters] Clean up null properties depending on conection mode

Fixes: https://github.com/elastic/kibana/issues/156354

## Summary
When previewing the request made to ES when creating a remote cluster config, when using proxy mode it will show null properties of sniff mode and when using sniff mode it will show null properties of proxy mode. This PR makes sure that those properties are only populated in the serialization function depending on the config.mode.

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to the remote clusters UI and create a new cluster config
3. Verify that with a proxy mode we dont see null properties of sniff mode, and viseversa.
4. Verify that creating the request success correctly, the remote cluster config is added, and that can be edited right after.


<details>
<summary>Screenshots</summary>

![Screenshot 2023-07-24 at 14 56 43](https://github.com/elastic/kibana/assets/1191206/32df2f9a-4c11-4c32-90ad-6b6893076505)
![Screenshot 2023-07-24 at 14 56 55](https://github.com/elastic/kibana/assets/1191206/cd955d7d-0212-4ac5-8ea2-6158d2eb867d)

</details>
